### ISSUE TITLE
tmux -CC: the wire gets no row, the bank gets a band row, and the control-mode leak is closed

### ISSUE THREAD

**Issue by overseers-desk** [overseers-desk/robco-term#33]
## Two things about `tmux -CC` change here

Typing `tmux -CC` in a channel turns that channel's pty into tmux's control-mode wire. Two faults follow, and both are addressed in one piece of work: a live leak, and a design the owner has now decided to replace. They are separable — Part A stands on its own and is the floor; Parts B–D are the redesign the owner settled through a design session (decisions recorded below).

---

## Part A — Control-mode protocol leaks onto the glass (the floor)

**Symptom.** After `tmux -CC` attaches, tmux's line protocol (`%output …`, `%begin`/`%end`, octal-escaped `\033[…`) prints as visible text on the glass of the channel that ran it. Reproduced against the pinned rio-vt fork (`Cargo.toml:37`, rev `b6b5a8e`); also against the registry copy of rio-vt 0.5.24, byte-identical on this path.

**Cause.** `Session::pump` (`crates/term/src/session.rs:341`) and `SshChannel::pump` (`crates/term/src/ssh_channel.rs:135`) hand every read chunk to **two** parsers on the same bytes:

```rust
self.dcs.feed(&chunk);
self.processor.advance(&mut self.term, &chunk);   // the grid
```

`crates/term/src/dcs.rs:13` asserts rio-vt swallows a DCS body whole, so the grid never sees envelope bytes. That premise is false. rio-vt's `advance_dcs_passthrough` leaves DCS state on `0x18`, `0x1A`, `0x1B` and `0x9C`. All four ride the tmux envelope as ordinary payload — the pane bootstrap runs `capture-pane -peqJ` (`crates/tmux-cc/src/command.rs:153`), whose `-e` puts raw `ESC [ … m` in every reply — so on the first capture the grid's parser drops to Ground and prints the rest of the protocol.

**Second route, independent of rio-vt.** `open_find` writes `\x1b[0m…` into the on-air session through `Session::feed` (`crates/term/src/session.rs:730`), which bypasses the DCS parser. Its `ESC` knocks the grid parser out of the envelope the same way.

**Fix (single-authority, no seam).** Collapse the dual-feed. The `DcsParser` already runs its own rio-vt parser to find the envelope and knows, byte by byte, which bytes are inside it. Make it the *only* thing that feeds the grid:

- `DcsParser::feed` gains an out-parameter (a `&mut Vec<u8>` the caller drains, or a small callback): it appends the **out-of-envelope** bytes to it and swallows in-envelope bytes. The chunk that opens the envelope is split at `ESC P 1000 p`; the chunk that closes it is split after `ST`.
- `Session::pump` / `SshChannel::pump` stop calling `self.processor.advance` directly. They do `grid_buf.clear(); self.dcs.feed(&chunk, &mut grid_buf); self.processor.advance(&mut self.term, &grid_buf);`.
- `Session::feed` and `SshChannel::feed` early-return while `self.dcs.tap().in_control_mode()` — a local injection has no business reaching the grid while the envelope is open. This closes the `open_find` route without leaning on any gateway guard (see Part B, where those guards are removed).
- Correct the false-premise docs, forward-only, to state the real rule (DcsParser is the single feeder; the grid sees only out-of-envelope bytes): `crates/term/src/dcs.rs:13-16` and `:119-125`; `crates/term/src/tmux_cc.rs:15-19` and `:145-150`.

**Missing test.** `a_dcs_block_reaches_the_tap_and_stays_off_the_grid` (`crates/term/tests/suite/transcript.rs:501`) uses `NoopTap`, so it never enters the `Envelope` state and does not exercise the bug. Add a `ControlModeTap` case (unit-level against a grid, or PTY-level) that feeds an envelope containing raw `ESC [ … m` (a `capture-pane -e` reply) and asserts the grid text is empty. Add an app-level assertion in `crates/app/tests/suite/tmux_flow.rs` that the carrier channel's own grid is clean after attach and after detach surfaces it — no such assertion exists today (phases only check bank/channel identity).

---

## Part B — The wire gets no row of its own (the model)

**Today.** `Channels::attach` (`crates/app/src/channels.rs:840`) *transports* the channel that ran `tmux -CC` to slot 1 of a new bank as a frozen "gateway" row, holding its home slot dark. Enter on that row detaches; every other key is swallowed; the glass shows the shell's last picture. The wire is treated as a screen.

**Decision (Tabula rasa).** The wire is not a screen and gets no row — the rule `channels.rs:76` already states for SSH ("no gateway row, because the wire has no on-screen carrier"). Instead:

- The channel that ran `tmux -CC` **stays in its home bank**, marked unavailable while the envelope is open, and comes back live when it closes (detach / `%exit` / gateway death). Its pty *is* the wire.
- A **new bank** is created for the attachment's tmux windows, filling **from slot 1** (no gateway occupying it).
- `Manager::Tmux` keeps a reference to the carrier's `(bank, channel)` in its home bank (reuse the existing `home` field, but it now points at a *live* carrier row, not a held-dark placeholder).

**Work:**

- `attach` (`channels.rs:834`): stop mutating the row's `bank`/`channel`/`title`. Push the new bank; record the carrier's `(bank, channel)` on `Manager::Tmux`; mark the carrier unavailable. The carrier does not move.
- The wire plumbing is hard-wired to "the row at `(new_bank, channel == 1)`" in two places — repoint both to the carrier's stored home location: `start_gateway` (`crates/app/src/window/tmux.rs:120`, takes `control_writer()`) and `pump_gateway` (`crates/app/src/window/tmux.rs:234-239`, the one `take_body()` call site).
- `open_tmux_window` (`channels.rs:556`): fill from slot 1. Remove the "gateway holds 1, windows from 2" logic (`channels.rs:557`, `first_free`'s held-slot handling `channels.rs:343-359`).
- `collapse_bank` (`channels.rs:961`) / detach: drop the new bank's window rows and remove the bank. There is no "row comes home" any more — the carrier never left; on collapse it is simply made available again and `leave_control_mode()` is called on its session (the Part A change makes its grid resume live with no protocol residue). `gateway_died` (`channels.rs:661`): if the carrier's own pty is alive, revive it; if the carrier process itself died, its home-bank row goes dark like any dead shell.
- `close_channel` (`channels.rs:624`): remove the `is_gateway` special-case; closing a tmux *window* row is tmux's to confirm as today, but detach is no longer "close the gateway row".
- `anchored_rows` (`channels.rs:677`): the carrier anchors as an ordinary home/pty row; drop the "gateway rows whose bank has a home" clause.
- Remove `is_gateway` (`channels.rs:258`) and `gateway_title` (`channels.rs:1047`) becomes the tmux band identity (Part C).
- **Band identity source.** Home bank: the local hostname. tmux bank: `tmux -CC # @<host>` (today's `gateway_title`, fits 32 chars). SSH bank: `user@host` (exists at `channels.rs:904`).

**Removed with the gateway row:** `gateway_key` (`crates/app/src/window/tmux.rs:413`, Enter-detaches); `is_gateway_on_air` (`crates/app/src/window/keys.rs:138`); the `type_bytes`/`write` gateway guards (`keys.rs:128`, `crates/app/src/window/mod.rs:1936`); and `frozen_glass` end to end — `PointerContext::frozen_glass` (`crates/term/src/pointer.rs:57`), its consumers (`marks_selection`, `marking`, the middle-button swallow in `on_press`), and where it is set from `is_gateway_on_air` (`crates/app/src/window/pointer.rs:194`). The unavailable carrier is simply not on the air and not selectable; its own input is already dropped by `drop_input_in_control_mode` (`session.rs:465`).

**Coordination.** Session `make-link-clickable` (branch `make-link-clickable`, feature: Ctrl+click opens URLs / OSC 8) reads `pointer_context()` for its `on_press` change but adds no new reads of `frozen_glass`. Removing `frozen_glass` is a one-line follow-through at `pointer_context()` for whoever lands second. Message that session before touching `crates/app/src/window/pointer.rs`. Keep `modifiers_from` (`keys.rs`) under its name.

**Tests to rewrite (all in `channels.rs`, ~18):** every tmux test encodes slot-1-gateway / transport / homecoming — `attaching_transports_the_channel_to_slot_one_of_a_new_bank` (`:1246`), `an_attachments_windows_fill_from_slot_two_and_the_first_takes_the_air` (`:1266`), `a_volunteered_window_landing_back_on_slot_two_does_not_take_the_air` (`:1290`), `a_collapse_brings_the_gateway_home_to_the_slot_it_never_gave_up` (`:1396`), `a_tmux_typed_on_an_ssh_channel_transports_and_comes_home_to_its_bank` (`:1410`), `a_store_never_crosses_banks_and_a_gateway_never_moves` (`:1464`), `a_close_of_a_window_row_is_tmuxs_to_do_and_the_row_stays_until_it_says` (`:1479`), and the rest. New assertions: carrier stays home and unavailable; windows fill from slot 1; detach makes the carrier available with a clean grid.

---

## Part C — The band row (the furniture)

**Decision.** Every bank page carries one **unnumbered** row at the top — the band row — drawn as a strip (LED or tape kit) reading the bank's identity (Part B). Numbered channel rows below read 01..N as today, so **one fewer numbered row fits per page**. The virtual slot space is unchanged (`CHANNEL_CAP = 99`); only the on-screen page loses a row.

**The band row is a label, never a channel.** It is not Alt+digit reachable, never takes the air, and is not a highlightable/selectable slot. Pressing its strip does not select anything (see Part D).

**Per-shell lane rule.** The shell declares whether the band row may reclaim the numeral lane. `ShellMetrics` gains `band_reclaims_lane: bool` (or equivalent):

- **annunciator** (`band_reclaims_lane = true`): lane is digits only; the band strip starts at `content_x` (wider by `numeral_width + column_gap = 46 + 16`, i.e. more characters).
- **slide rule** (`true`): same; lane is digits only (`36 + 8`).
- **switchboard** (`false`): the lane is the switch plate, well and lever. Keep the band strip where the numbered rows' strips are; stamp the band name across the plate lane where the digits go. The plate is 170px wide, ample; the existing 2-digit stamp is small and centred, so this needs a new "band label" paint path in `switchboard::row_furniture` rather than reusing the numeral stamp.

**The strip's x lives in one place** — `strip_rect` (`crates/chassis/src/furniture.rs:311`) — so the reclaim rule is a per-row branch there, not a per-theme constant. A band strip that starts further left is a longer character count (`width_for_units` of the wider span), varying by shell and kit; that is fine.

**Work:**

- `BankStrips` (`crates/chassis/src/strip.rs:122`): add the band row as a distinct optional element (e.g. `band: Option<BandStrip { title: String, reclaims_lane: bool }>`) rather than folding it into the numbered `rows` — the numbered rows keep their `label`/`numeral`/`channel` invariants clean. `StripRow` already tolerates an empty numeral (`annunciator.rs:209`, `slide_rule.rs:179`, `switchboard.rs:234` all guard `!numeral.is_empty()`), which the switchboard path can lean on, but the band label text is a new argument regardless.
- `BankPager::strips` (`crates/app/src/bank.rs:170`): build the band from the bank's identity; number the channel rows 1..(rows_on_page − 1).
- `rows_visible`: reserve one row of headroom for the band. Decide whether that is `top_padding` growth or one reserved row in the pitch count; `BankGeometry::rows_visible` (`crates/chassis/src/bank.rs`) and its call site `settle_rows` (`crates/app/src/window/mod.rs:1420`) carry the arithmetic.
- `furniture::bank_pieces` (`crates/chassis/src/furniture.rs:389`): draw the band row above the numbered rows, honouring `band_reclaims_lane` for its `strip_rect`; the numbered rows draw as today at their (shifted-down) indices.
- `strip_at` (`furniture.rs:327`) and the press routing (`crates/app/src/window/seam.rs:61`): the band row's rectangle routes to the Part D press behaviour, not to `press_strip`.
- **Local hostname source** needs picking (the `hostname` crate, `uname`, or `/etc/hostname`). Flagged below.

**Tests to update (row-count / piece-count / strip-x / rows_visible):** `the_shipped_appliance_puts_a_plate_and_a_strip_per_row_on_the_casting` (`furniture.rs:747`, asserts `pieces.len() == 1 + 4 + 2*rows + 1` and `strip_x = content_x + 46 + 16`), `the_switchboard_has_no_plate_and_stamps_tape` (`furniture.rs:835`), `the_numerals_restart_at_one_on_every_page` (`bank.rs:266`), `a_reflow_that_keeps_the_row_count_keeps_the_hand_picked_page` (`bank.rs:404`, "240px holds three rows"), `crates/chassis/tests/suite/bank_frame_geometry.rs::rows_per_page_at_sampled_window_heights`, and `crates/app/tests/suite/bank_chrome.rs` (`1 + 4 + 2*rows + 1`). The numbered-row assertions stay; the band row adds its own pieces and consumes one row of headroom.

---

## Part D — The band-switch card, the detach card, and the detach key

**Decision 1 — band-switch card.** Crossing into a bank raises the overlay `Notice` with the bank's identity, riding the degauss that already fires. In `step_bank` (`crates/app/src/window/bank.rs:130`), after `select_bank(landed)`, call `self.notice.raise(<bank identity>, now)`. `Notice` (`crates/app/src/overlay.rs:170`): 0.8 opacity, 3s hold, 200ms fade — a glance, then gone.

**Decision 2 — detach is a key, pressing the band row shows how.** A press must not detach: click-that-makes-something-disappear is poor UI, and the band row is not a selectable channel, so it cannot ride the channel-close key. Pressing the band row's strip on a managed bank raises a card naming the detach key (reuse `Notice`: `notice.raise("detach: Alt+Shift+D", now)`). On the home bank the band row press does nothing (nothing to detach).

**Decision 3 — the detach key is `Alt+Shift+D`.** Its own dedicated key, acting on the whole tmux attachment on view (not a channel) — detach is a bank operation, so it sits in the Alt family (Alt+PgUp/Dn steps bands, Alt+digit dials a slot), away from the channel keys (Ctrl+Shift+C/V/W/T). The tmux session keeps running; reattach with `tmux -CC attach`. Add it to `shortcut_key` (`crates/app/src/window/keys.rs:205`), mapping to detach on the tmux bank in view → `Close::Detach` → `gateway.detach()`.

---

## Verification

- `cargo test -p robco-term --lib` (dcs, tmux_cc unit tests, incl. the new ControlModeTap grid-clean case)
- `cargo test -p robco-term --test suite` (transcript)
- `cargo test -p robco-app --test suite` (channel_bank, bank_chrome, bank_furniture, tmux_flow — needs `/usr/bin/tmux`, present at 3.5a)
- `cargo test -p robco-chassis` (furniture, bank, strip) and `cargo test -p robco-chassis --test suite` (bank_frame_geometry)
- Manual: run the app, type `tmux -CC`. Confirm the home-bank carrier row goes unavailable with no protocol text on its glass; a new band appears with windows from slot 01; the band row reads the host; pressing the band row raises the detach card; `Alt+Shift+D` detaches; the carrier revives clean. Then `tmux -CC` over SSH and repeat.

## Invariants

Holds: the band row is bank-column furniture, not a session cell; the detach and band-switch cards are `Notice` overlays over the glass, which the invariant already permits (a copy of the screen still yields only what the session sent). Unconfigured tmux stays indistinguishable from a build without it — no band-switch card, no carrier-unavailable state, ever fires without an attachment.

## Open defaults (flagged, non-blocking — adopt unless the owner says otherwise)

1. **Carrier row while attached:** shown dimmed/locked in its home bank; pressing it does nothing. (Alternative: press jumps to the tmux band it drives.)
2. **SSH bank + `Alt+Shift+D`:** detach/close the SSH connection, mirroring tmux. (Or no-op, leaving SSH close on its current path.)
3. **Local hostname source** for the home band label: the `hostname` crate vs `uname` vs `/etc/hostname`.
4. **tmux band label text:** `tmux -CC # @<host>` (today's form) vs a shorter `@<host>`.

## Provenance

Design settled in a Claude Code session with the owner. Full map of every touched symbol with file:line is in this body; no conversation context is required to execute. Related: the leak was found while the owner pasted control-mode text off the glass and asked why it rendered.
